tm2/pkg/bft/config/config.go

```diff
@@ -40,37 +41,62 @@ func DefaultConfig() *Config {
 	}
 }
 
-type ConfigOptions func(cfg *Config)
+type Option func(cfg *Config)
 
 // LoadOrMakeConfigWithOptions loads configuration or saves one
 // made by modifying the default config with override options
-func LoadOrMakeConfigWithOptions(root string, options ConfigOptions) (*Config, error) {
-	var cfg *Config
-
-	configPath := join(root, defaultConfigFilePath)
+func LoadOrMakeConfigWithOptions(root string, opts ...Option) (*Config, error) {
+	// Initialize the config as default
+	var (
+		cfg        = DefaultConfig()
+		configPath = join(root, defaultConfigFilePath)
+	)
+
+	// Check if the config exists
 	if osm.FileExists(configPath) {
-		var loadErr error
-
 		// Load the configuration
-		if cfg, loadErr = LoadConfigFile(configPath); loadErr != nil {
+		loadedCfg, loadErr := LoadConfigFile(configPath)
+		if loadErr != nil {
 			return nil, loadErr
 		}
 
-		cfg.SetRootDir(root)
-		cfg.EnsureDirs()
-	} else {
-		cfg = DefaultConfig()
-		if options != nil {
-			options(cfg)
+		// Merge the loaded config with the default values
+		if err := mergo.Merge(loadedCfg, cfg); err != nil {
+			return nil, err
 		}
-		cfg.SetRootDir(root)
-		cfg.EnsureDirs()
-		WriteConfigFile(configPath, cfg)
 
-		// Validate the configuration
-		if validateErr := cfg.ValidateBasic(); validateErr != nil {
-			return nil, fmt.Errorf("unable to validate config, %w", validateErr)
+		// Set the root directory
+		loadedCfg.SetRootDir(root)
+
+		// Make sure the directories are initialized
+		if err := loadedCfg.EnsureDirs(); err != nil {
+			return nil, err
```

Comment:
Short answer: should be good, we only call this method when initializing the node configuration (once in `gnoland start`, and other few times in tests).

The existence of this method, let alone what it does, is...not ideal.

Just to give you context on _why_ we need to "ensure directories" in the first place, because I thought it was also confusing, given we have the `RootDir` as part of each configuration TOML signature.

We didn't actually save the `RootDir` field when we marshal the TOML, in the previous template approach:
https://github.com/gnolang/gno/blob/6112ffd31f68131fdf07bc61408ddd32e5d6f38b/tm2/pkg/bft/config/toml.go#L64

that's why we had the `SetRootDir` step when loading up the configuration, and ensuring it's there with `EnsureDirs`.

I plan to completely refactor the config flow we have now for node initializations, this minor fix was just to keep it in line with previous legacy functionality